tests/gtfs/test_routes.py

```diff
@@ -125,3 +135,119 @@ def test_lookup_is_stable(self):
+            (
+                pathlib.Path("tests/data/newport-20230613_gtfs.zip"),
+                pytest.raises(
+                    ValueError,
+                    match=r"`path` expected file extension .pkl. "
+                    r"Found .zip",
+                ),
+            ),
+            (
+                here("tests/data/test_file.pkl"),
+                pytest.raises(
+                    FileNotFoundError,
+                    match=re.escape(
+                        f"{here('tests/data/test_file.pkl')} not "
+                        "found on file."
+                    ),
+                ),
+            ),
+        ],
+    )
+    def test_get_saved_route_type_lookup_raises(
+        self, path: Union[str, pathlib.Path], expected: Type[RaisesContext]
+    ):
+        """Test raises."""
+        with expected:
+            get_saved_route_type_lookup(path=path)
+
+    @pytest.mark.parametrize(
+        "pkl_name, test_obj, expected",
+        [
+            # invalid object once .pkl unserialized
+            (
+                "list_pkl.pkl",
+                [1, 2, 3, 4, 5],
+                pytest.raises(
+                    TypeError,
+                    match=re.escape(
+                        "Serialized object in specified .pkl file is of type: "
+                        "<class 'list'>. Expected (<class 'dict'>, "
+                        "<class 'pandas.core.frame.DataFrame'>)"
+                    ),
+                ),
+            ),
+            # empty dataframe
+            (
+                "empty_pkl.pkl",
+                pd.DataFrame({"test_col": []}),
+                pytest.warns(
+                    UserWarning, match="Route type lookup has length of 0"
+                ),
+            ),
+            # df with invalid columns
+            (
+                "invalid_col_pkl.pkl",
+                pd.DataFrame(
+                    {"route_type": [0, 1], "route_desc": ["car", "bus"]}
+                ),
+                pytest.warns(
+                    UserWarning,
+                    match=(
+                        "Unexpected column 'route_desc' in route type lookup"
+                    ),
+                ),
+            ),
+        ],
+    )
+    def test_get_saved_route_type_lookup_invalid_pkl(
+        self, tmp_path, pkl_name, test_obj, expected
+    ):
+        """Test defences when unserialized pickle is invalid obj type."""
+        out_pth = os.path.join(tmp_path, pkl_name)
+        _create_pkl(obj=test_obj, out_pth=out_pth)
+        with expected:
+            get_saved_route_type_lookup(out_pth)
+
+    def test_get_saved_route_type_lookup_on_pass(self, tmp_path):
+        """Test get_saved_route_type_lookup_on_pass()."""
+        found_lkp = get_saved_route_type_lookup()
+        assert isinstance(found_lkp, pd.DataFrame), ".pkl did not return df"
+        assert len(found_lkp) == 91, ".pkl is an unexpected length"
```

Comment:
I can't guarantee this won't change in the future. May want to remove it?

There is a test for the stability of the scraped table, though it is an integration test and not run as part of the default suite (eg `pytest --runinteg`)

Reply:
Yes it would probably be best to change the length check. I don't mind though as it's an easy update if we ever change the route lookup